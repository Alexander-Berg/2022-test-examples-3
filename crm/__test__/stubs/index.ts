export * from './account';
export * from './tabs';
