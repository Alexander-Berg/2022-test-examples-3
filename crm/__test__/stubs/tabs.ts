export const tabs = {
  data: {
    tabs: [
      {
        tabId: 1,
        tabCaption: 'Кредиты',
        blocks: {
          '1': {
            componentType: 'Table',
            title: 'AccountCredits',
            url: '/v0/blocks/accountCredits?accountId=9955466',
          },
          '3': {
            componentType: 'Table',
            title: 'AccountDebts',
            url: '/v0/blocks/accountDebts?accountId=9955466',
          },
        },
        layout: {
          schema: { rows: [{ areaId: '1', blockWidthRatio: 1 }] },
          areaIdToBlocksIds: { '1': ['1', '3'] },
        },
      },
    ],
  },
};
