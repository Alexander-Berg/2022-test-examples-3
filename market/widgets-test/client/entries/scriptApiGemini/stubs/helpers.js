export const createEntityLongName = (n, name, delimiter) => {
    const parts = Array(n).fill(name);

    return parts.join(n % 2 ? ' ' : delimiter);
};

export const createWarnings = ({warningCode, warningText}) => {
    if (warningCode) {
        return [
            {
                code: warningCode,
                message: warningText,
            },
        ];
    }

    return [];
};
