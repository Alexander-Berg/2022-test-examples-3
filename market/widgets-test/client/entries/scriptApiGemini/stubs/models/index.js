import {compose, identity} from 'ramda';

import WidgetTypes from 'src/widgets-main/common/constants/widgetTypes';
import {setStateWidgetThemeId} from 'src/widgets-main/common/specHelpers/state';
import {Creator as MultiTypeWidgetCreator} from 'src/widgets-create/client/helpers/widgetsCreator/concrete/multiType';

import {withStubs} from '../../helpers/withStubs';
import {createInitialState} from './state.fixtures';

const processInitialParams = ({params}) => {
    const {themeId, themeRows} = params;

    const processedParams = {
        ...params,
    };

    if (themeId) {
        processedParams.themeId = Number(themeId);
    }

    if (themeRows) {
        processedParams.themeRows = Number(themeRows);
    }

    return processedParams;
};

const dataProvider = params => {
    const {themeRows, warningCode, warningText, themeId, offersCount, modelsCount} = params;

    const data = compose(
        themeId ? setStateWidgetThemeId(themeId) : identity,
        createInitialState({
            themeRows,
            warningCode,
            warningText,
            offersCount,
            modelsCount,
        }),
    )();

    return {
        type: WidgetTypes.Models,
        data,
    };
};

export default withStubs({dataProvider, processInitialParams}, MultiTypeWidgetCreator);
