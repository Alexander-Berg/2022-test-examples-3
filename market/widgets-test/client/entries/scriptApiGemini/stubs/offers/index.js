import {compose, identity} from 'ramda';

import WidgetTypes from 'src/widgets-main/common/constants/widgetTypes';
import {setStateWidgetThemeId} from 'src/widgets-main/common/specHelpers/state';
import {Creator as MultiTypeWidgetCreator} from 'src/widgets-create/client/helpers/widgetsCreator/concrete/multiType';

import {withStubs} from '../../helpers/withStubs';
import {createInitialState} from './state.fixtures';

const processInitialParams = ({params}) => {
    const {themeId} = params;

    const processedParams = {
        ...params,
    };

    if (themeId) {
        processedParams.themeId = Number(themeId);
    }

    return processedParams;
};

const dataProvider = params => {
    const {warningCode, warningText, outOfStock, themeId} = params;

    const data = compose(
        themeId ? setStateWidgetThemeId(themeId) : identity,
        createInitialState({warningCode, warningText, outOfStock}),
    )();

    return {
        type: WidgetTypes.Offers,
        data,
    };
};

export default withStubs({dataProvider, processInitialParams}, MultiTypeWidgetCreator);
