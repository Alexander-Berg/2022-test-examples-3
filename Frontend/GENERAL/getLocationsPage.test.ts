import { groupItems, simplifyCityServices, transformCityToEntityListItem, getLocationsPage } from './getLocationsPage';

import { IServicesListItemCity, ICitiesListItem, IListWithPagination } from '../../typings/femida';
import { ILcRootTurboJSON, LcJobsEntityListItem } from '../../typings/lpc';

import { getTruncatedCitiesItems, getCities } from '../stubs/femidaCities';
import { getEntityListSection } from '../stubs/lpcEntityListSection';

const getError = (result: ILcRootTurboJSON) => {
    // @ts-ignore
    return result.content[0].content[0].error as string;
};

describe('getLocationsPage', () => {
    let entityListSection: ILcRootTurboJSON;
    let citiesListItems: ICitiesListItem[];
    let cities: IListWithPagination<ICitiesListItem>;

    const expectedCityItems: LcJobsEntityListItem[] = [
        {
            name: 'Гродно',
            publicationsCount: 2,
            group: {
                title: 'Белоруссия',
                data: {
                    countryCode: 'BY',
                    description: undefined,
                    position: undefined,
                    slug: undefined,
                },
            },
            services: [{ name: 'Яндекс Go', slug: 'go', priority: 2 }],
            slug: 'grodno',
        },
        {
            name: 'Москва',
            publicationsCount: 123,
            group: {
                title: 'Россия',
                data: {
                    countryCode: 'RU',
                    description: 'some description',
                    position: 123,
                    slug: 'russia',
                },
            },
            services: [],
            slug: 'moscow',
        },
    ];

    beforeEach(() => {
        entityListSection = getEntityListSection();
        citiesListItems = getTruncatedCitiesItems();
        cities = getCities();
    });

    describe('simplifyCityServices', () => {
        let cityServices: IServicesListItemCity[];

        beforeEach(() => {
            cityServices = [
                {
                    id: 1,
                    name: 'Яндекс Go',
                    slug: 'go',
                    priority: 2,
                },
                {
                    id: 34,
                    name: 'Интранет',
                    slug: 'intranet',
                    priority: 1,
                },
                {
                    id: 35,
                    name: 'Маркет',
                    slug: 'market',
                    priority: 3,
                },
                {
                    id: 36,
                    name: 'Облако',
                    slug: 'cloud',
                    priority: 2,
                },
            ];
        });

        it('should preserve only needed fields of services', () => {
            expect(simplifyCityServices(cityServices)).toStrictEqual([
                { name: 'Яндекс Go', slug: 'go', priority: 2 },
                { name: 'Интранет', slug: 'intranet', priority: 1 },
                { name: 'Маркет', slug: 'market', priority: 3 },
                { name: 'Облако', slug: 'cloud', priority: 2 },
            ]);
        });

        it('should preserve empty array if passed that', () => {
            expect(simplifyCityServices([])).toEqual([]);
        });
    });

    describe('transformCityToEntityListItem', () => {
        it('should properly handle city entity', () => {
            expect(transformCityToEntityListItem(citiesListItems[0])).toStrictEqual(expectedCityItems[0]);
        });

        it('should extract optional parameters: description, position and slug from country of city', () => {
            expect(transformCityToEntityListItem(citiesListItems[1])).toStrictEqual(expectedCityItems[1]);
        });
    });

    describe('groupItems', () => {
        it('should handle each city item', () => {
            expect(groupItems(citiesListItems)).toStrictEqual(expectedCityItems);
        });

        it('should properly handle invoking without params', () => {
            expect(groupItems()).toEqual([]);
        });
    });

    it('should add error if cities is not passed', () => {
        const results = [
            getLocationsPage(entityListSection),
            getLocationsPage(entityListSection, {} as IListWithPagination<ICitiesListItem>),
        ];

        for (const result of results) {
            const error = getError(result);

            // Ensure, that we have that error field in result.
            expect(typeof error).toBe('string');
        }
    });

    it('should correct extend entity list section with cities', () => {
        expect(getLocationsPage(entityListSection, cities)).toMatchSnapshot();
    });
});
