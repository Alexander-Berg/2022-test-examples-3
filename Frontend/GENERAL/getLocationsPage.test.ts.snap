// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLocationsPage should correct extend entity list section with cities 1`] = `
Object {
  "adv": true,
  "bug": Object {
    "data-text": "",
    "id": "10031339.26ca2126fd58cdcae360592e6e659247800836f8",
  },
  "canonical": "",
  "content": Array [
    Object {
      "block": "lc-page",
      "content": Array [
        Object {
          "anchor": "jobs-spisok-lokacijservisovprofessij-1",
          "block": "lc-jobs-entity-list",
          "content": Array [
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "KZ",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Казахстан",
              },
              "name": "Астана",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "astana",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Люберцы",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "lubertsy",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Красноярск",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "krasnoyarsk",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Краснодар",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "krasnodar",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Калуга",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "kaluga",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Ижевск",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "izhevsk",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Челябинск",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "chelyabinsk",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "BY",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Белоруссия",
              },
              "name": "Гродно",
              "publicationsCount": 2,
              "services": Array [
                Object {
                  "name": "Яндекс Go",
                  "priority": 2,
                  "slug": "go",
                },
              ],
              "slug": "grodno",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Омск",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "omsk",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "US",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "США",
              },
              "name": "Ньюберипорт",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "newburyport",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Ивантеевка",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "ivanteevka",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Мытищи",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "mytishchi",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Сасово",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "sasovo",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Удалённая работа",
              "publicationsCount": 11,
              "services": Array [
                Object {
                  "name": "Яндекс Go",
                  "priority": 5,
                  "slug": "go",
                },
                Object {
                  "name": "Интранет",
                  "priority": 2,
                  "slug": "intranet",
                },
                Object {
                  "name": "Облако",
                  "priority": 4,
                  "slug": "cloud",
                },
              ],
              "slug": "remote",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "CN",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Китай",
              },
              "name": "Шанхай",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "shanghai",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "CH",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Швейцария",
              },
              "name": "Цюрих",
              "publicationsCount": 4,
              "services": Array [
                Object {
                  "name": "Яндекс Go",
                  "priority": 3,
                  "slug": "go",
                },
                Object {
                  "name": "Облако",
                  "priority": 1,
                  "slug": "cloud",
                },
              ],
              "slug": "zurich",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "CH",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Швейцария",
              },
              "name": "Люцерн",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "lucerne",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "CN",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Китай",
              },
              "name": "Тайбэй",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "taipei",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "IL",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Израиль",
              },
              "name": "Тель-Авив",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "tel-aviv",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "FI",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Финлядния",
              },
              "name": "Хельсинки",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "helsinki",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Нижний Новгород",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "nizhniy-novgorod",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "BY",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Белоруссия",
              },
              "name": "Минск",
              "publicationsCount": 4,
              "services": Array [
                Object {
                  "name": "Интранет",
                  "priority": 2,
                  "slug": "intranet",
                },
                Object {
                  "name": "Маркет",
                  "priority": 1,
                  "slug": "market",
                },
                Object {
                  "name": "Тулзы",
                  "priority": 1,
                  "slug": "tools",
                },
              ],
              "slug": "minsk",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "UA",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Украина",
              },
              "name": "Киев",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "kiyv",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Владимир",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "vladimir",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Новгород",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "novgorod",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Ростов-на-Дону",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "rostov-na-dony",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Казань",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "kazan",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Новосибирск",
              "publicationsCount": 0,
              "services": Array [],
              "slug": "novosibirsk",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Екатеринбург",
              "publicationsCount": 3,
              "services": Array [
                Object {
                  "name": "Яндекс Go",
                  "priority": 1,
                  "slug": "go",
                },
                Object {
                  "name": "Интранет",
                  "priority": 1,
                  "slug": "intranet",
                },
                Object {
                  "name": "Маркет",
                  "priority": 1,
                  "slug": "market",
                },
              ],
              "slug": "ekaterinburg",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Санкт-Петербург",
              "publicationsCount": 8,
              "services": Array [
                Object {
                  "name": "Яндекс Go",
                  "priority": 2,
                  "slug": "go",
                },
                Object {
                  "name": "Интранет",
                  "priority": 1,
                  "slug": "intranet",
                },
                Object {
                  "name": "Маркет",
                  "priority": 3,
                  "slug": "market",
                },
                Object {
                  "name": "Облако",
                  "priority": 2,
                  "slug": "cloud",
                },
              ],
              "slug": "saint-petersburg",
            },
            Object {
              "group": Object {
                "data": Object {
                  "countryCode": "RU",
                  "description": undefined,
                  "position": undefined,
                  "slug": undefined,
                },
                "title": "Россия",
              },
              "name": "Москва",
              "publicationsCount": 43,
              "services": Array [
                Object {
                  "name": "Яндекс Go",
                  "priority": 17,
                  "slug": "go",
                },
                Object {
                  "name": "Интранет",
                  "priority": 7,
                  "slug": "intranet",
                },
                Object {
                  "name": "Маркет",
                  "priority": 8,
                  "slug": "market",
                },
                Object {
                  "name": "Облако",
                  "priority": 8,
                  "slug": "cloud",
                },
                Object {
                  "name": "sdfs",
                  "priority": 2,
                  "slug": "testservice",
                },
                Object {
                  "name": "Тулзы",
                  "priority": 1,
                  "slug": "tools",
                },
              ],
              "slug": "moscow",
            },
          ],
          "emptyListProps": Object {
            "background": Object {
              "customPositions": Object {
                "horizontal": Object {
                  "unit": "px",
                  "value": 0,
                },
                "vertical": Object {
                  "unit": "px",
                  "value": 0,
                },
              },
              "customSizes": Object {
                "height": Object {
                  "unit": "px",
                  "value": 0,
                },
                "proportions": false,
                "width": Object {
                  "unit": "px",
                  "value": 0,
                },
              },
              "fixed": false,
              "hasImmersiveAnimation": false,
              "image": Object {
                "url": "",
              },
              "positions": "center_center",
              "sizes": "cover",
              "type": "image",
              "video": Object {
                "mp4Url": "",
                "type": "uploaded",
                "webmUrl": "",
              },
            },
            "events": Array [],
            "header": Object {
              "background": Object {
                "customPositions": Object {
                  "horizontal": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "vertical": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "customSizes": Object {
                  "height": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "proportions": false,
                  "width": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "fixed": false,
                "hasImmersiveAnimation": false,
                "image": Object {
                  "url": "",
                },
                "positions": "center_center",
                "sizes": "cover",
                "type": "image",
                "video": Object {
                  "mp4Url": "",
                  "type": "uploaded",
                  "webmUrl": "",
                },
              },
              "color": "#000",
              "content": "320235320270321207320265320263320276 320275320265 320275320260321210320273320270 320277320276 321202320260320272320276320274321203 320267320260320277321200320276321201321203",
              "events": Array [],
              "jobsOffsets": Object {
                "sidePadding": "none",
              },
              "padding": Object {
                "bottom": "none",
                "top": "none",
              },
              "size": "l",
              "tag": "",
              "type": "header",
            },
            "image": Object {
              "height": 1242,
              "mimeType": "image/png",
              "url": "https://avatars.mds.yandex.net/get-lpc/1368426/05d40da5-f987-430f-9e98-9dcd730ce896/orig",
              "width": 1242,
            },
            "imageAlt": "320257320275320264320265320272321201",
            "imagePosition": "left",
            "jobsOffsets": Object {
              "sidePadding": "none",
            },
            "link": Object {
              "events": Array [],
              "openInNewTab": false,
              "text": "",
              "type": "url",
              "url": "",
              "withSmoothScroll": false,
            },
            "linkText": Object {
              "background": Object {
                "customPositions": Object {
                  "horizontal": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "vertical": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "customSizes": Object {
                  "height": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "proportions": false,
                  "width": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "fixed": false,
                "hasImmersiveAnimation": false,
                "image": Object {
                  "url": "",
                },
                "positions": "center_center",
                "sizes": "cover",
                "type": "image",
                "video": Object {
                  "mp4Url": "",
                  "type": "uploaded",
                  "webmUrl": "",
                },
              },
              "color": "#000",
              "content": "320243320267320275320260321202321214 320261320276320273321214321210320265",
              "events": Array [],
              "jobsOffsets": Object {
                "sidePadding": "none",
              },
              "padding": Object {
                "bottom": "none",
                "top": "none",
              },
              "size": "l",
              "tag": "",
              "type": "text",
            },
            "padding": Object {
              "bottom": "none",
              "top": "none",
            },
            "text": Object {
              "background": Object {
                "customPositions": Object {
                  "horizontal": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "vertical": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "customSizes": Object {
                  "height": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "proportions": false,
                  "width": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "fixed": false,
                "hasImmersiveAnimation": false,
                "image": Object {
                  "url": "",
                },
                "positions": "center_center",
                "sizes": "cover",
                "type": "image",
                "video": Object {
                  "mp4Url": "",
                  "type": "uploaded",
                  "webmUrl": "",
                },
              },
              "color": "#999",
              "content": "320233320270320261320276 320262321213 320270321211320265321202320265 321207321202320276-321202320276 321201320277320265321206320270321204320270321207320265321201320272320276320265, 320273320270320261320276 320277320270321210320265321202320265 320275320260 320264321200321203320263320276320274 321217320267321213320272320265. 320237321200320276320262320265321200321214321202320265, 320277320276320266320260320273321203320271321201321202320260, 320265321211321221 321200320260320267 320272320276321200321200320265320272321202320275320276321201321202321214 320262320262320276320264320260 320267320260320277321200320276321201320260.",
              "events": Array [],
              "jobsOffsets": Object {
                "sidePadding": "none",
              },
              "padding": Object {
                "bottom": "none",
                "top": "none",
              },
              "size": "l",
              "tag": "",
              "type": "text",
            },
            "view": "normal",
          },
          "errorListProps": Object {
            "background": Object {
              "customPositions": Object {
                "horizontal": Object {
                  "unit": "px",
                  "value": 0,
                },
                "vertical": Object {
                  "unit": "px",
                  "value": 0,
                },
              },
              "customSizes": Object {
                "height": Object {
                  "unit": "px",
                  "value": 0,
                },
                "proportions": false,
                "width": Object {
                  "unit": "px",
                  "value": 0,
                },
              },
              "fixed": false,
              "hasImmersiveAnimation": false,
              "image": Object {
                "url": "",
              },
              "positions": "center_center",
              "sizes": "cover",
              "type": "image",
              "video": Object {
                "mp4Url": "",
                "type": "uploaded",
                "webmUrl": "",
              },
            },
            "events": Array [],
            "header": Object {
              "background": Object {
                "customPositions": Object {
                  "horizontal": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "vertical": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "customSizes": Object {
                  "height": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "proportions": false,
                  "width": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "fixed": false,
                "hasImmersiveAnimation": false,
                "image": Object {
                  "url": "",
                },
                "positions": "center_center",
                "sizes": "cover",
                "type": "image",
                "video": Object {
                  "mp4Url": "",
                  "type": "uploaded",
                  "webmUrl": "",
                },
              },
              "color": "#000",
              "content": "320235320265 321201320274320276320263320273320270 320267320260320263321200321203320267320270321202321214 321201320277320270321201320276320272",
              "events": Array [],
              "jobsOffsets": Object {
                "sidePadding": "none",
              },
              "padding": Object {
                "bottom": "none",
                "top": "none",
              },
              "size": "l",
              "tag": "",
              "type": "header",
            },
            "image": Object {
              "height": 1242,
              "mimeType": "image/png",
              "url": "https://avatars.mds.yandex.net/get-lpc/1368426/05d40da5-f987-430f-9e98-9dcd730ce896/orig",
              "width": 1242,
            },
            "imageAlt": "320257320275320264320265320272321201",
            "imagePosition": "right",
            "jobsOffsets": Object {
              "sidePadding": "none",
            },
            "link": Object {
              "events": Array [],
              "openInNewTab": false,
              "text": "",
              "type": "url",
              "url": "",
              "withSmoothScroll": false,
            },
            "linkText": Object {
              "background": Object {
                "customPositions": Object {
                  "horizontal": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "vertical": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "customSizes": Object {
                  "height": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "proportions": false,
                  "width": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "fixed": false,
                "hasImmersiveAnimation": false,
                "image": Object {
                  "url": "",
                },
                "positions": "center_center",
                "sizes": "cover",
                "type": "image",
                "video": Object {
                  "mp4Url": "",
                  "type": "uploaded",
                  "webmUrl": "",
                },
              },
              "color": "#000",
              "content": "320243320267320275320260321202321214 320261320276320273321214321210320265",
              "events": Array [],
              "jobsOffsets": Object {
                "sidePadding": "none",
              },
              "padding": Object {
                "bottom": "none",
                "top": "none",
              },
              "size": "l",
              "tag": "",
              "type": "text",
            },
            "padding": Object {
              "bottom": "none",
              "top": "none",
            },
            "text": Object {
              "background": Object {
                "customPositions": Object {
                  "horizontal": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "vertical": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "customSizes": Object {
                  "height": Object {
                    "unit": "px",
                    "value": 0,
                  },
                  "proportions": false,
                  "width": Object {
                    "unit": "px",
                    "value": 0,
                  },
                },
                "fixed": false,
                "hasImmersiveAnimation": false,
                "image": Object {
                  "url": "",
                },
                "positions": "center_center",
                "sizes": "cover",
                "type": "image",
                "video": Object {
                  "mp4Url": "",
                  "type": "uploaded",
                  "webmUrl": "",
                },
              },
              "color": "#999",
              "content": "320234321213 320277321200320276320261320276320262320260320273320270, 320275320276 321203 320275320260321201 320275320265 320277320276320273321203321207320270320273320276321201321214. 320237320276320277321200320276320261321203320271321202320265 320277320265321200320265320267320260320263321200321203320267320270321202321214 321201321202321200320260320275320270321206321203, 320277321200320276320262320265321200321214321202320265 320275320260320273320270321207320270320265 320270320275321202320265321200320275320265321202320260, 320270 320265321201320273320270 320276321210320270320261320272320260 320277320276320262321202320276321200320270321202321201321217, 320275320260320277320270321210320270321202320265 320275320260320274 320262 <a href='https://forms.yandex.ru/surveys/10031549.21a902f6166ba06f34f92596983bf7aa6e186bc4/'>320277320276320264320264320265321200320266320272321203</a>.",
              "events": Array [],
              "jobsOffsets": Object {
                "sidePadding": "none",
              },
              "padding": Object {
                "bottom": "none",
                "top": "none",
              },
              "size": "s",
              "tag": "",
              "type": "header",
            },
            "view": "normal",
          },
          "events": Array [],
          "isExpanded": true,
          "key": "1f9f7b8d-6a34-4276-8a93-d5ba47a18fd3",
          "mix": Array [
            "lpc-section",
          ],
          "offsets": Object {
            "padding": Object {
              "bottom": "none",
              "top": "none",
            },
          },
          "sectionId": "1f9f7b8d-6a34-4276-8a93-d5ba47a18fd3",
          "type": "locations",
          "useMocks": false,
          "withOffsets": false,
        },
      ],
      "preview": false,
    },
  ],
  "css": "
.alert,
.modal,
.page__result,
.page__container,
.sticky_fixed .sticky__content {
	width: 100%;
	max-width: none !important;
}
",
  "description": "Все вакансии Яндекса",
  "fallbackDisabled": true,
  "favicon": "https://avatars.mds.yandex.net/get-lpc/1674605/85ae003a-c76f-42ae-81bc-853e93846d7b/orig",
  "keywords": "",
  "noindex": false,
  "ograph": Object {
    "description": "Вакансии Яндекса",
    "locale": "ru",
    "site_name": "Работа в Яндексе",
    "title": "Вакансии Яндекса",
    "type": "website",
    "url": "https://hamster.yandex.ru/jobs/professions/mob-app-developer",
  },
  "page_type": "turbo-lpc",
  "preconnectDomains": Array [
    "https://mc.yandex.ru/",
    "https://mc.admetrica.ru/",
  ],
  "referrer-policy": "no-referrer-when-downgrade",
  "service": "turbo-lpc",
  "title": "Все вакансии Яндекса",
  "url": "https://yandex.ru/jobs/components/withvacancies",
  "yandexVerification": "jfdbvkhdfbvhkjdfbvk",
}
`;
